log4j-core/src/main/java/org/apache/logging/log4j/core/appender/rolling/RollingFileManager.java

```diff
@@ -506,9 +506,16 @@ public synchronized void rollover() {
 
             if (rollover(rolloverStrategy)) {
                 try {
-                    size = 0;
-                    initialTime = System.currentTimeMillis();
                     createFileAfterRollover();
+                    final File file = new File(getFileName());
+                    size = file.length();
+                    try {
+                        final FileTime creationTime = (FileTime) Files.getAttribute(file.toPath(), "creationTime");
+                        initialTime = creationTime.toMillis();
+                    } catch (Exception ex) {
+                        LOGGER.warn("Unable to get current file time for {}", file);
+                        initialTime = System.currentTimeMillis();
+                    }
```

Comment:
- Even if the rollover fails, I would still reset the `size` field. This is the field used by [`SizeBasedTriggeringPolicy`](https://logging.apache.org/log4j/3.x/manual/appenders/rolling-file.html#SizeBasedTriggeringPolicy) to determine, when a rollover is due. Otherwise the triggering policy might attempt a rollover at **each** log event. It is safe to assume that the new rollover might fail too.
- The `initialTime` field is not really used by any triggering policy at runtime. It is only used at **configuration time** and never queried again. I would leave it as is.